appservice/src/tryGetSiteResource.ts

```diff
@@ -28,11 +28,19 @@ export async function tryGetWebAppSlot(client: WebSiteManagementClient, resource
  * 3. { "Code": "NotFound", "Message": "Server farm with name appsvc_linux_centralus not found." }
  */
 async function tryGetSiteResource<T>(callback: () => Promise<T | WebSiteManagementModels.DefaultErrorResponse>): Promise<T | undefined> {
-    const result: T | WebSiteManagementModels.DefaultErrorResponse = await callback();
-    const regExp: RegExp = /NotFound/i;
-    if (regExp.test(parseError(result).errorType) || ('error' in result && regExp.test(parseError(result.error).errorType))) {
-        return undefined;
-    } else {
-        return <T>result;
+    try {
+        const result: T | WebSiteManagementModels.DefaultErrorResponse = await callback();
+        const regExp: RegExp = /NotFound/i;
+        if (regExp.test(parseError(result).errorType) || ('error' in result && regExp.test(parseError(result.error).errorType))) {
```

Comment:
This code was already checking for the error `NotFound` which makes me think that this API doesn't throw an error, but returns a `DefaultErrorResponse`.  Is that not the case with the `AuthorizationFailed` errors?

Reply:
You are right, it does return DefaultErrorResponse, but when checked for this particular scenario "The client 'xxxxxx@xxx.com' with object id 'XXXXXXXXXXXXXXXX' does not have..." is thrown rather than been returned.  I wasn't sure why that was the case for this particular error.